src/intro_movie_scenes_lib.c

```diff
@@ -0,0 +1,528 @@
+    int winIn = 0x1F;
+    IntroMovieCircleWipeEffect *effectData = &data->circleWipeEffect;
+    BOOL setBrightnessNeutral = 0;
+    PMLCDTarget screenId = PM_LCD_BOTTOM;
+    int winOut = 0;
+    effectData->active = TRUE;
+    effectData->finished = FALSE;
+    effectData->duration = duration;
+    effectData->counter = 0;
+    effectData->whichScreen = isTopScreen;
+    effectData->x = FX16_ONE;
+    effectData->y = 0;
+    effectData->kind = kind;
+    int bgId;
+    int color;
+    if (isTopScreen) {
+        bgId = GF_BG_LYR_MAIN_0;
+        screenId = PM_LCD_TOP;
+    } else {
+        bgId = GF_BG_LYR_SUB_0;
+        screenId = PM_LCD_BOTTOM;
+    }
+    if (effectData->kind == 0 || effectData->kind == 1) {
+        color = RGB_WHITE;
+    } else {
+        color = RGB_BLACK;
+    }
+    BG_SetMaskColor(bgId, color);
+    if (effectData->kind == 0 || effectData->kind == 2) {
+        winIn = 0;
+        winOut = 0x1F;
+        setBrightnessNeutral = TRUE;
+    }
+    IntroMovie_SetBgWindowsPosition(0, 0, 255, 192, isTopScreen);
+    IntroMovie_WindowsOn_SetInsideOutsidePlanes(winIn, winOut, 1, 1, isTopScreen);
+    if (setBrightnessNeutral) {
+        SetMasterBrightnessNeutral(screenId);
+    }
+    effectData->task = SysTask_CreateOnMainQueue(Task_IntroMovie_CircleWipeEffect, effectData, 0);
+    Main_SetHBlankIntrCB(IntroMovie_CircleWipeEffect_HBlankCB, effectData);
+}
+
+BOOL IntroMovie_WaitCircleWipeEffect(IntroMovieOvyData *data) {
+    IntroMovieCircleWipeEffect *effectData = &data->circleWipeEffect;
+    if (!effectData->active) {
+        return TRUE;
+    }
+    if (effectData->finished) {
+        return TRUE;
+    }
+    return FALSE;
+}
+
+void Task_IntroMovie_CircleWipeEffect(SysTask *task, void *pVoid) {
+    IntroMovieCircleWipeEffect *effectData = (IntroMovieCircleWipeEffect *)pVoid;
+
+    u16 idx = FX_DEG_TO_IDX(180 * FX32_ONE) * effectData->counter / effectData->duration;
+    effectData->x = FX_CosIdx(idx);
+    effectData->y = FX_SinIdx(idx);
+    ++effectData->counter;
+    if (effectData->counter >= effectData->duration) {
+        SysTask_Destroy(effectData->task);
+        effectData->task = NULL;
+        effectData->finished = TRUE;
+        effectData->active = FALSE;
+
+        u8 screen = effectData->whichScreen == 0 ? PM_LCD_BOTTOM : PM_LCD_TOP;
+        if (effectData->kind == 1) {
+            sub_0200FBF4(screen, RGB_WHITE);
+        } else if (effectData->kind == 3) {
+            sub_0200FBF4(screen, RGB_BLACK);
+        } else {
+            if (effectData->whichScreen) {
+                GX_SetVisibleWnd(0);
+            } else {
+                GXS_SetVisibleWnd(0);
+            }
+            SetMasterBrightnessNeutral(screen);
+        }
+        Main_SetHBlankIntrCB(NULL, NULL);
+    }
+}
+
+void IntroMovie_CircleWipeEffect_HBlankCB(void *pVoid) {
+    IntroMovieCircleWipeEffect *data = (IntroMovieCircleWipeEffect *)pVoid;
+
+    int vcount = GX_GetVCount();
+    if (vcount > 0xBF) {
```

Comment:
arbritrary number?

Reply:
starting from the 192nd scanline, we're in vblank

Reply:
can we make a constant for this?

Reply:
out-of-scope imo

Reply:
fair, though it is something that *should* be done eventually